### PR TITLE
EgressIPs: Allow moving IP

### PR DESCRIPTION
**Pull request by dulek**
**- What this PR does and why is it needed**
Anticipating openshift/cloud-network-config-controller#95 this allows support for operations when CloudPrivateIPConfig is moved from one node to another skipping the usual release-assign dance.

Without this commit new IP is only added to the EgressIP CRD status without deleting the old one, resulting in controller deleting the CloudPrivateIPConfig completely in the next iteration of reconclieEgressIP.

**- Special notes for reviewers**
I extended the cases where CloudPrivateIP controller selects IPs to delete to include a case when `Status.Node` is different on old and new resource.

**- How to verify it**
I verified it with OpenShift env running openshift/cloud-network-config-controller#95

**- Description for the changelog**
Support MovePrivateIP operations on CloudPrivateIPConfig.

### CODE REVIEW

**Comment by andreaskaris** [comment]
This /lgtm as it totally makes sense to me but I will let @kyrtapz chime in as I have to admit that I don't fully seize all of the stuff that's going on in that function after that change

**Review by andreaskaris — APPROVED** [review]

**Comment by kyrtapz** [comment]
Just for clarity, this change is only required for cloud providers that implement moving an IP from one node to another in one step ([currently only openstack](https://github.com/openshift/cloud-network-config-controller/pull/95)).
For other cloud providers CNCC will first remove the IP from the old node and set the status to empty which will result in `shouldDelete = shouldDelete && newCloudPrivateIPConfig.Status.Node == ""` being true. After that it will update the status once the IP gets assigned to the new node: https://github.com/openshift/cloud-network-config-controller/pull/95/files#diff-8cf9eee1c396d93e41e0ce10915b91fd632a2e35e78274ac9972a31057ad0428R503-R517

@dulek does this sound right?

File: go-controller/pkg/ovn/egressip.go [hunk 1]
```diff
@@ -579,7 +579,8 @@ func (oc *DefaultNetworkController) reconcileCloudPrivateIPConfig(old, new *ocpc
 			ocpcloudnetworkapi.CloudPrivateIPConfigConditionType(newCloudPrivateIPConfig.Status.Conditions[0].Type) == ocpcloudnetworkapi.Assigned &&
 			kapi.ConditionStatus(newCloudPrivateIPConfig.Status.Conditions[0].Status) == kapi.ConditionTrue
 		// See above explanation for the delete
```
Comment (kyrtapz) [inline go-controller/pkg/ovn/egressip.go:581]: Can you update the explanation mentioned here to include the introduced change?

Comment (dulek) [reply go-controller/pkg/ovn/egressip.go:583]: Done!

**Comment by dulek** [comment]
@dcbw, can you take a look at this PR? Thanks!

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm

**Comment by dcbw** [comment]
Any way to add/update the testcases for this?

**Pull request merged**